app/models/repository.rb

```diff
@@ -321,9 +325,9 @@ def self.create_from_hash(repo_hash)
       g.assign_attributes repo_hash.slice(*Repository::API_FIELDS)
 
       if g.changed?
-        return g.save ? g : nil
```

Comment:
Fun fact. In the latest version of Rails, returning early in a transaction will roll the transaction back. If I have time I would like to comb through the rest of the code in Libraries to see if we are doing this in other places.